config/snapshot/config_snapshot_test.go

```diff
@@ -108,6 +108,96 @@ func TestObserve_OnInit(t *testing.T) {
 	}
 }
 
+func TestObserve_OnInitRetry(t *testing.T) {
+	tests := []struct {
+		name             string
+		config           *configpb.Configuration
+		configGeneration int64
```

Comment:
I don't think "configGeneration" is actually a parameter of this test. Consider just hard-coding it when constructing your fake.

Reply:
Done, thanks!